evm-adapters/src/fuzz.rs

```diff
@@ -87,6 +87,14 @@ impl<'a, S, E: Evm<S>> FuzzedExecutor<'a, E, S> {
                     .call_raw(self.sender, address, calldata.clone(), 0.into(), false)
                     .expect("could not make raw evm call");
 
+                // When assume cheat code is triggered return a special string "FOUNDRY::ASSUME"
+                if returndata.as_ref() == ASSUME_MAGIC_RETURN_CODE {
+                    let _ = return_reason.borrow_mut().insert(reason);
+                    let err = "ASSUME: Too many global rejects";
+                    let _ = revert_reason.borrow_mut().insert(err.to_string());
```

Comment:
This is a hacky thing. Actual here we revert the test case. For this to work the revert_reason doesn't matter. However when the test case fails for "Too many global rejects" reason, revert_reason is printed to console instead of the actual error. Hence we put it here.